src/_pytest/pytester.py

```diff
@@ -1370,7 +1391,13 @@ def runpytest_subprocess(self, *args, timeout: Optional[float] = None) -> RunRes
         if plugins:
             args = ("-p", plugins[0]) + args
         args = self._getpytestargs() + args
-        return self.run(*args, timeout=timeout)
+        # Use a consistent encoding, isolated from the host system.
+        encoding = "utf-8"
+        env = {
+            **os.environ.copy(),
+            "PYTHONIOENCODING": encoding,
+        }
```

Comment:
this makes it hard/impossible to test under other encoding scenarios.

Reply:
Can add an `env` argument to `runpytest_subprocess`, but I prefer to punt on this as it doesn't seem like it was necessary up to now.

Reply:
actually on second thought I don't think we should do this at all (but perhaps provide `encoding` as an option).  it potentially masks a class of common encoding errors by waving away the default behaviour

Reply:
It masks configuration issues on the host machine, not a bug in the tested program. I think such isolation is generally good. Another reason is to have consistency with inprocess tests (where the output is captured, and stdout/stderr are always utf-8).

But just to be clear, the current behavior is definitely wrong, it just *assumes* UTF-8. In the PR message I described an alternative approach (use `locale.getpreferredencoding()`) and why I rejected it, do you prefer that anyway? (or another approach)

Reply:
hmmmm yeah that's tricky

I think what I'd do is grown an `encoding=...` parameter and default it to `None`, pass it along if non-None as PYTHONIOENCODING, and decode using either `getpreferredencoding()` (nothing passed) or `encoding` if passed?

I think that would be the most ~realistic setup (so the testsuite doesn't have some surprising behaviour that's different from reality)

Reply:
I'm fine with this plan, but would prefer the default to be `utf-8`, not `None`. In this case, I prefer the tests to be consistent/isolated over being realistic.

Reply:
my fear is that someone will assume pytest is following the default behaviour, write a test against the pytest behaviour, and then be surprised when it's actually broken in reality